openlibrary/templates/type/edition/view.html

```diff
@@ -500,18 +500,32 @@ <h4>$:_('Links <span class="gray small sansserif">outside Open Library</span>')<
         $# pages like /books/ia:foo00bar are fake records created from metadata API.
         $# Adding them to lists doesn't work. Disabling it to avoid any issues.
         $if "lists" in ctx.features and not page.is_fake_record():
-          <h2 class="list-heading"><a href="$page.url()/lists">$_('Lists containing this Book')</a></h2>
+          <div class="lists-heading">
+            <h2 class="lists-title">$_('Lists')</h2>
+            $if ctx.user:
+              <a href="$page.url()/lists">$_('See All')</a>
+            $else:
+              <a href="$homepath()/lists">$_('Learn More')</a>
+          </div>
           <div class="user-book-options">
             <div class="Tools tools-override">
-              $if work:
-                $ component_times['lists widget: WorkListTime'] = time()
-                $:render_template("lists/widget", work, include_header=False, include_widget=False)
-                $ component_times['lists widget: WorkListTime'] = time() - component_times['lists widget: WorkListTime']
-
-              $if edition and is_editionless_work:
-                $ component_times['lists widget: EditionListTime'] = time()
-                $:render_template("lists/widget", edition, include_header=False, include_widget=False)
-                $ component_times['lists widget: EditionListTime'] = time() - component_times['lists widget: EditionListTime']
+              $if work or edition:
+                $ has_lists = False
+                $if work:
+                  $if len(work.get_lists(sort=False)) > 0:
+                    $ has_lists = True
+                    $ component_times['lists widget: WorkListTime'] = time()
+                    $:render_template("lists/widget", work, include_header=False, include_widget=False)
+                    $ component_times['lists widget: WorkListTime'] = time() - component_times['lists widget: WorkListTime']
+                $if edition and is_editionless_work:
+                  $if len(edition.get_lists(sort=False)) > 0:
+                    $ has_lists = True
+                    $ component_times['lists widget: EditionListTime'] = time()
+                    $:render_template("lists/widget", edition, include_header=False, include_widget=False)
+                    $ component_times['lists widget: EditionListTime'] = time() - component_times['lists widget: EditionListTime']
+
+                $if not has_lists:
+                  $_('This work does not appear on any lists.')
```

Comment:
After reverting the list widget rendering code, please ensure that this if statement has the same indentation level as the preceding `$if work` and `$if edtition and is_editionless_work` statements.